src/components/picker/types.tsx

```diff
@@ -238,7 +239,7 @@ export interface PickerItemProps {
    * Callback for onPress action
    * @param selected true\false in multi mode and undefined in single mode
    */
-  onPress?: (selected?: boolean) => void;
+  onPress?: (selected: boolean | undefined, props: any) => void | Promise<boolean>;
```

Comment:
I would add the note you added there about returning false to disable closing the picker